[Issue #327] GUI: Angular GUI - Work in Progress

Jimmy-VVang opened the pull request:
install node and npm: https://docs.npmjs.com/getting-started/installing-node
cd to textdb/textdb-angular-gui/
run 'npm install'
run 'npm start'

Added operators bar and drag and drop functionality
Modified .gitignore

chenlica commented:
@Jimmy-VVang Please remove your name from the subject.  Add screenshots if you can.  Who should be the reviewers?

Jimmy-VVang commented:
Added pop-up functionality to drag and drop
moved drag and drop from flowchart component to operator-bar component
separated data to be accessed by other components
![capture1](https://cloud.githubusercontent.com/assets/18473185/23483624/96c07b04-fe88-11e6-9c90-a50bf9f425b3.PNG)

zuozhiw commented:
The PRs about the new GUI are all merged. This WIP PR can be closed.
